src/components/VRating/VRating.ts

```diff
@@ -0,0 +1,181 @@
+      type: String,
+      default: 'secondary'
+    },
+    color: {
+      type: String,
+      default: 'primary'
+    },
+    emptyIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingEmpty'
+    },
+    fullIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingFull'
+    },
+    halfIcon: {
+      type: String,
+      default: '$vuetify.icons.ratingHalf'
+    },
+    halfIncrements: Boolean,
+    length: {
+      type: Number,
+      default: 5
+    },
+    readonly: Boolean,
+    showHover: Boolean,
+    value: {
+      type: Number,
+      required: true
+    }
+  },
+
+  data: () => ({
+    hoverIndex: -1
+  }),
+
+  computed: {
+    isHovering (): boolean {
+      return this.showHover && this.hoverIndex >= 0
+    }
+  },
+
+  methods: {
+    isHalfEvent (e: MouseEvent): boolean {
+      if (this.halfIncrements) {
+        const rect = e.target && (e.target as HTMLElement).getBoundingClientRect()
+        if (rect && e.offsetX < rect.width / 2) return true
+      }
+
+      return false
+    },
+    createClickFn (i: number): Function {
+      return (e: MouseEvent) => {
+        if (this.readonly) return
+
+        if (this.isHalfEvent(e)) this.$emit('input', i + 0.5)
+        else this.$emit('input', i + 1)
+      }
+    },
+    createProps (i: number): ItemSlotProps {
+      const click = this.createClickFn(i)
+      const isHovered = Math.floor(this.hoverIndex) > i
+      const isHalfHovered = this.halfIncrements && !isHovered && (this.hoverIndex - i) % 1 > 0
+      const isFilled = Math.floor(this.value) > i
+      const isHalfFilled = this.halfIncrements && !isFilled && (this.value - i) % 1 > 0
+
+      return {
+        index: i,
+        value: this.value,
+        isFilled,
+        isHalfFilled,
+        isHovered,
+        isHalfHovered,
+        click
+      }
+    },
+    getIconName (props: ItemSlotProps): string {
+      let iconName
+
+      if (this.isHovering) {
+        iconName = props.isHovered ? this.fullIcon : props.isHalfHovered ? this.halfIcon : this.emptyIcon
+      } else {
+        iconName = props.isFilled ? this.fullIcon : props.isHalfFilled ? this.halfIcon : this.emptyIcon
+      }
+
+      return remapInternalIcon(this, iconName)
+    },
+    getColor (props: ItemSlotProps): string {
+      if (this.isHovering) {
+        if (props.isHovered || props.isHalfHovered) return this.color
+      } else {
+        if (props.isFilled || props.isHalfFilled) return this.color
+      }
+
+      return this.backgroundColor
+    },
+    onMouseEnter (e: MouseEvent, i: number): void {
+      if (this.isHalfEvent(e)) this.hoverIndex = i + 0.5
+      else this.hoverIndex = i + 1
+    },
+    onMouseLeave (e: MouseEvent): void {
+      this.hoverIndex = -1
+    },
+    genItem (h: CreateElement, i: number): VNode | VNodeChildrenArrayContents | string {
+      const props = this.createProps(i)
+
+      if (this.$scopedSlots.item) return this.$scopedSlots.item(props)
+
+      let listeners = {
```

Comment:
`let listeners: Record<string, function> = {`

Reply:
`<string, EventListener>`?

Reply:
MouseEvent isn't compatible with EventListener, only Event is.